File: src/main/java/se/lu/nateko/cp/meta/persistence/TripleIteratorDocumentSource.java
package se.lu.nateko.cp.meta.persistence;

import java.util.Iterator;

import org.openrdf.model.Statement;
import org.semanticweb.owlapi.model.IRI;
import org.semanticweb.owlapi.rio.RioMemoryTripleSource;

public class TripleIteratorDocumentSource extends RioMemoryTripleSource {

	private final IRI documentIRI;

	public TripleIteratorDocumentSource(IRI documentIri, Iterator<Statement> statements) {
		super(statements);
		this.documentIRI = documentIri;
	}

	@Override
	public IRI getDocumentIRI() {
		return documentIRI;
	}
}

File: src/main/scala/se/lu/nateko/cp/meta/persistence/TripleIteratorDocumentSource.scala
package se.lu.nateko.cp.meta.persistence

import scala.jdk.CollectionConverters.IteratorHasAsJava

import org.eclipse.rdf4j.model.Statement;
import org.semanticweb.owlapi.model.IRI;
import org.semanticweb.owlapi.rio.RioMemoryTripleSource;

class TripleIteratorDocumentSource(documentIri: IRI, statements: Iterator[Statement]) extends RioMemoryTripleSource(statements.asJava) {
	override def getDocumentIRI: IRI = documentIri
}
